src/App.jsx

```diff
@@ -1,18 +1,25 @@
-import { Component } from 'react';
 import './App.css';
-import MenuPage from 'pages/MenuPage.jsx';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import AppLayout from 'components/AppLayout/AppLayout.jsx';
 import { CartProvider } from 'contexts/CartContext.jsx';
+import { AuthProvider } from 'contexts/AuthContext.jsx';
+import ProtectedRoute from 'router/ProtectedRoute.jsx';
+import routes from 'router/routerConfig.js';
 
-class App extends Component {
-    render() {
-        return (
-            <CartProvider>
-                <div className="App">
-                    <MenuPage />
-                </div>
-            </CartProvider>
-        );
-    }
-}
+const App = () => (
+    <AuthProvider>
+        <CartProvider>
```

Comment:
To prevent nessesity to check isLoading state inside of multiple component it's better to do it somewhere here, and prevent rendering application befor you will know current user state